laba3Java/src/main/java/Turnstile.java

```diff
@@ -0,0 +1,69 @@
+ class Turnstile {
+    private CardCatalog catalog;
+    private int passPrise;
+    public Turnstile( CardCatalog Catalog){
+        catalog = Catalog;
+        passPrise = 8;
+    }
+    void LetGo(PassCard card){
+        System.out.print(ANSI_GREEN + " Пропустити." + ANSI_RESET);
+        if(card.GetIsDiscount()){ System.out.print(ANSI_PURPLE + "     Attention! Пільговий квиток." + ANSI_RESET);}
+         System.out.println("");
+        card.AddAccept();
+    }
+    void DontLetGo(PassCard card){
+        card.AddDeclines();
+    }
+  void  PassTheCard(PassCard card) {
+      if (card != null && catalog.IsInCatalog(card)) {
+          System.out.print(" ID: "+card.GetId());
+        if(card.GetIsActive()) {
+            if (card instanceof Card_OnMoneyAmount) {
```

Comment:
Specific check for every card type should be moved to the method in the appropriate card. Here you should call that method on the object of common type PassCard and get the correct result for a specific card using polymorphism.
This implementation with "if" and "instanceof" sequence is not optimal and can be rated at 12 out of 20.